### ISSUE TITLE
[Feature Request] Support setting `training=False` argument in `BeamSearchDecoder`

### ISSUE THREAD

**Issue by HanGuo97** [tensorflow/addons#364]
**System information**
- TensorFlow version (you are using): 2.0-beta
- TensorFlow Addons version: 0.40
- Is it in the tf.contrib (if so, where): No
- Are you willing to contribute it (yes/no): No
- Are you willing to maintain it going forward? (yes/no): No

**Describe the feature and the current behavior/state.**
Current `BeamSearchDecoder` does not support setting `training=False` argument during the computation. This limits many scenarios in which model behaves differently between training/inference stage, e.g., dropout.

**Will this change the current api? How?**
An extra argument that supports setting the `training` argument should be added to this [line](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/beam_search_decoder.py#L547)

**Who will benefit with this feature?**
Everyone who wants the RNN Cell to behave differently between training and inference (e.g., dropout).
**Any Other info.**

**Comment by seanpmorgan** [comment]
CC some people more familiar with seq2seq @qlzh727 @guillaumekln

**Comment by guillaumekln** [comment]
Yes, it seems a `training` flag should be forwarded from `BaseDecoder.call` to the cell call of concrete decoders (`BasicDecoder` and `BeamSearchDecoder`).
